Remotion/Validation/UnitTests/DefaultValidatorExtensionsTest.cs

```diff
@@ -1,14 +1,416 @@
+      addingPropertyValidationRuleBuilder.Setup(_ => _.SetValidator(It.IsAny<Func<PropertyValidationRuleInitializationParameters, IPropertyValidator>>()))
+          .Callback<Func<PropertyValidationRuleInitializationParameters, IPropertyValidator>>(func => createdValidator = func.Invoke(initParameters));
+
+      addingPropertyValidationRuleBuilder.Object.MinLength(17);
+
+      addingPropertyValidationRuleBuilder.Verify();
+      Assert.That(createdValidator, Is.InstanceOf<LengthValidator>());
+      var lengthValidator = (LengthValidator)createdValidator;
+      Assert.That(lengthValidator.Min, Is.EqualTo(17));
+      Assert.That(lengthValidator.Max, Is.Null);
+    }
+
+    [Test]
+    public void MaxLength_ForString_ReturnsLengthValidator ()
+    {
+      var validationMessage = new InvariantValidationMessage("Fake message");
+      var initParameters = new PropertyValidationRuleInitializationParameters(validationMessage);
+      IPropertyValidator createdValidator = null;
+      var addingPropertyValidationRuleBuilder = new Mock<IAddingPropertyValidationRuleBuilder<object, string>>();
+      addingPropertyValidationRuleBuilder.Setup(_ => _.SetValidator(It.IsAny<Func<PropertyValidationRuleInitializationParameters, IPropertyValidator>>()))
+          .Callback<Func<PropertyValidationRuleInitializationParameters, IPropertyValidator>>(func => createdValidator = func.Invoke(initParameters));
+
+      addingPropertyValidationRuleBuilder.Object.MaxLength(42);
+
+      addingPropertyValidationRuleBuilder.Verify();
+      Assert.That(createdValidator, Is.InstanceOf<LengthValidator>());
+      var lengthValidator = (LengthValidator)createdValidator;
+      Assert.That(lengthValidator.Min, Is.EqualTo(0));
+      Assert.That(lengthValidator.Max, Is.EqualTo(42));
+    }
+
+    [Test]
+    public void ExactLength_ForString_ReturnsLengthValidator ()
+    {
+      var validationMessage = new InvariantValidationMessage("Fake message");
+      var initParameters = new PropertyValidationRuleInitializationParameters(validationMessage);
+      IPropertyValidator createdValidator = null;
+      var addingPropertyValidationRuleBuilder = new Mock<IAddingPropertyValidationRuleBuilder<object, string>>();
+      addingPropertyValidationRuleBuilder.Setup(_ => _.SetValidator(It.IsAny<Func<PropertyValidationRuleInitializationParameters, IPropertyValidator>>()))
+          .Callback<Func<PropertyValidationRuleInitializationParameters, IPropertyValidator>>(func => createdValidator = func.Invoke(initParameters));
+
+      addingPropertyValidationRuleBuilder.Object.ExactLength(23);
+
+      addingPropertyValidationRuleBuilder.Verify();
+      Assert.That(createdValidator, Is.InstanceOf<LengthValidator>());
+      var lengthValidator = (LengthValidator)createdValidator;
+      Assert.That(lengthValidator.Min, Is.EqualTo(23));
+      Assert.That(lengthValidator.Max, Is.EqualTo(23));
```

Comment:
Hmm...I think we should also test that the initParameters are used correctly by adding the following assertion at the end:
```
Assert.That(lengthValidator .ErrorMessage, Is.SameAs(validationMessage));
```

We should do this for each method. 

The intention of this is to show that the parameter is used correctly and no one creates a hardcoded message inline in the extension method. It goes in the same direction as testing that the other arguments are passed correctly, such the Min and Max values here.